### ISSUE TITLE
CMake fails building v1.1.1

### ISSUE THREAD

**Issue by Luzifer** [lastpass/lastpass-cli#244]
```
dpkg-buildpackage
-----------------

dpkg-buildpackage: source package lastpass-cli
dpkg-buildpackage: source version 1.1.1-3
dpkg-buildpackage: source distribution xenial
 dpkg-source --before-build lastpass-cli-1.1.1
dpkg-buildpackage: host architecture i386
 fakeroot debian/rules clean
dh clean 
   dh_testdir
   dh_auto_clean
	make -j1 clean
make[1]: Entering directory '/<<PKGBUILDDIR>>'
mkdir -p build && cd build && cmake -DCMAKE_INSTALL_PREFIX:PATH=/usr -DCMAKE_INSTALL_MANDIR:PATH=/usr/share/man ..
CMake Error: The current CMakeCache.txt directory /<<PKGBUILDDIR>>/build/CMakeCache.txt is different than the directory /lastpass-cli/build where CMakeCache.txt was created. This may result in binaries being created in the wrong place. If you are not sure, reedit the CMakeCache.txt
CMake Error: The source "/<<PKGBUILDDIR>>/CMakeLists.txt" does not match the source "/lastpass-cli/CMakeLists.txt" used to generate cache.  Re-run cmake with a different source directory.
Makefile:29: recipe for target 'build/Makefile' failed
make[1]: *** [build/Makefile] Error 1
make[1]: Leaving directory '/<<PKGBUILDDIR>>'
dh_auto_clean: make -j1 clean returned exit code 2
debian/rules:16: recipe for target 'clean' failed
make: *** [clean] Error 2
dpkg-buildpackage: error: fakeroot debian/rules clean gave error exit status 2
--------------------------------------------------------------------------------
Build finished at 20170105-1950
```

I'm not sure about this but as the build was triggered from a clean tree and between 1.1.0 and 1.1.1 was an edit changing CMake things this might be a bug now… (If not I'm requesting help with what I'm doing wrong here…)

**Comment by bcopeland** [comment]
Hmm, where does the `<<PKGBUILDDIR>>` come from?

I guess Makefile clean target should remove build/ directory.  But I'm not replicating this when building from a clean checkout.

**Comment by Luzifer** [comment]
Thats something from the Launchpad build system. Whole log here: https://launchpadlibrarian.net/301352093/buildlog_ubuntu-xenial-i386.lastpass-cli_1.1.1-3_BUILDING.txt.gz

**Comment by bcopeland** [comment]
If you make this change to the toplevel makefile, does it get any better?
```
diff --git a/Makefile b/Makefile
index aacc8a1..2ed62df 100644
--- a/Makefile
+++ b/Makefile
@@ -9,6 +9,7 @@ all: $(CMAKEMAKE)
 
 clean: $(CMAKEMAKE)
        $(MAKE) -C $(BUILDDIR) clean
+       $(RM) -r $(BUILDDIR)/*
 
 doc-man: $(CMAKEMAKE)
        $(MAKE) -C $(BUILDDIR) doc-man
```

**Comment by bcopeland** [comment]
[I edited this to remove the `/*` part, in case BUILDDIR is defined-empty to avoid tears.]

**Comment by bcopeland** [comment]
I plan to cut a new release shortly - can you test that Makefile change so I can decide whether to include it?

**Comment by Luzifer** [comment]
Yep, that did work… 

[BuildLog](https://launchpadlibrarian.net/303560353/buildlog_ubuntu-xenial-amd64.lastpass-cli_1.1.1-4_BUILDING.txt.gz)

Full applied diff (excluding changelog)

```diff
diff --git a/Makefile b/Makefile
index aacc8a1..2195a6d 100644
--- a/Makefile
+++ b/Makefile
@@ -9,6 +9,7 @@ all: $(CMAKEMAKE)

 clean: $(CMAKEMAKE)
        $(MAKE) -C $(BUILDDIR) clean
+       $(RM) -r $(BUILDDIR)

 doc-man: $(CMAKEMAKE)
        $(MAKE) -C $(BUILDDIR) doc-man
diff --git a/debian/control b/debian/control
index e0132d9..926f779 100644
--- a/debian/control
+++ b/debian/control
@@ -2,7 +2,7 @@ Source: lastpass-cli
 Section: utils
 Priority: optional
 Maintainer: Troy Heber <troyh@debian.org>
-Build-Depends: debhelper (>= 9), quilt (>= 0.47), libssl-dev, libxml2-dev, libcurl4-openssl-dev, asciidoc, xsltproc, docbook-xsl
+Build-Depends: debhelper (>= 9), quilt (>= 0.47), libssl-dev, libxml2-dev, libcurl4-openssl-dev, asciidoc, xsltproc, docbook-xsl, cmake, pkg-config
 Standards-Version: 3.9.8.0

 Package: lastpass-cli
```

**Comment by bcopeland** [comment]
Thanks, I merged this.

**Comment by Luzifer** [comment]
Thanks for the fix! The 1.1.2 build worked fine in first try… :)
